cross-compilation incorrectly searches the host for cmake find_* commands

For #2277 my arm cross-compile builds were claiming that zlib was available, but it was the host x86 zlib, and:

```
arm-debug-internal-64: **** 6 build errors ****
	/work/dr/git/src/clients/drcachesim/reader/compressed_file_reader.h:38:18: fatal error: zlib.h: No such file or directory
```

We should revisit the CMAKE_FIND_ROOT_PATH settings in our toolchain files: it looks like they are set to search the host dirs for find_*(), which does not seem the right thing to do.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1749,16 +1749,10 @@ function (link_with_pthread target)
 endfunction ()
 
 # zlib and snappy are used for some clients/ and tests.
-if (CMAKE_CROSSCOMPILING)
-  # find_package looks at the host system.
-  # XXX i#2285: we should re-visit the CMAKE_FIND_ROOT_PATH settings!
-  set(ZLIB_FOUND OFF)
-else ()
-  find_package(ZLIB)
-  # On Ubuntu 14.10, 32-bit builds fail to link with -lsnappy, just ignore.
-  if (UNIX AND X64)
-    find_library(libsnappy snappy)
-  endif ()
+find_package(ZLIB)
+# On Ubuntu 14.10, 32-bit builds fail to link with -lsnappy, just ignore.
+if (UNIX AND X64)
+  find_library(libsnappy snappy)
 endif ()
 
 if (BUILD_CLIENTS)

